### ISSUE TITLE
A question about neg log IBP function

### ISSUE THREAD

**Issue by Hzfinfdu** [McGill-NLP/polytropon#4]
Hi, thank you for your impressive idea. It's very instructive for me. 

Not sure if I have mistaken the neg log IBP function. I thought it should be a binary matrix indicating whether this task-skill pair is activated. But in line 78 in polytropon.py 
`aux_loss = [self.neg_log_IBP(p) for n, p in self.model.named_parameters() if "skill_logits" in n]`
it seems that this function takes a non-binary matrix as input. 

This conflict kind of confuses me. Do I need to convert it to binary?

Thanks!

**Comment by ducdauge** [comment]
Hi,

Thank you for your question! As you correctly pointed out, the task-skill allocation matrix is binary as it is a collection of Bernoulli samples. However, to make it end-to-end differentiable, I use a continuous relaxation (the Concrete distribution). 

I believe that there is currently a bug in the code in that the input to `neg_log_IBP` should be squeezed between [0, 1] via a sigmoid and then binarised via thresholding. I will fix this bug now, so thanks for pointing that out.

**Comment by Hzfinfdu** [comment]
Thank you for your kind response!
